packages/react/test/provider.spec.tsx

```diff
@@ -136,5 +139,74 @@ describe('OpenFeatureProvider', () => {
         await waitFor(() => expect(screen.queryByText('👍')).toBeInTheDocument(), { timeout: DELAY * 2 });
       });
     });
+
+    describe('useMutateContext', () => {
```

Comment:
These tests look great but could you please add one more for the following setup?

```
<OpenFeatureProvider domain={DOMAIN1}>
  <React.Suspense fallback={<div>{FALLBACK}</div>}>
    <TestComponent name="Will" />
    <OpenFeatureProvider>
      <React.Suspense fallback={<div>{FALLBACK}</div>}>
        <TestComponent name="Todd" />
      </React.Suspense>
    </OpenFeatureProvider>
  </React.Suspense>
</OpenFeatureProvider>
```

The nested component should update the global context.

Reply:
@beeme1mr I have set this up. In this scenario, it will update both TestComponents. Is that the expected behaviour?

Reply:
I'd like a test that ensures that updating context in the nested component updates the global context and not the domain scoped context. The other component probably isn't necessary in the test.

Reply:
The reason I'm asking is it seems like something is wrong, shouldn't domain scoped context be unchanged even if the global context changes?

Reply:
Yes, you're correct. Is there something wrong with my example?

Reply:
Yes when I set the globally scoped context thats nested under the scoped one, I was expecting "Will says hi" to stay unchanged, but instead I'm getting both "Todd says aloha" and "Will says aloha".

Maybe its out of scope of this PR to narrow down if its the react implementation or the core lib with the issue, but wanted to confirm it was unintended behaviour.

Reply:
Ah, okay. I understand the issue. Basically, domain-scoped context is only managed separately if it's explicitly set for the domain. Otherwise, it inherits from the global context. The logic is that most of the time the context should be the same but we wanted to give users the ability to override this behavior if necessary.

I'm now wondering how we should handle this in the hooks. I think I'll need to test it out myself tomorrow before I can provide a solid recommendation.

Thanks for your effort on this.

Reply:
See: https://github.com/open-feature/js-sdk/pull/1031/files#r1801753490

Reply:
@toddbaert adding a property to set global or not doesn't solve the issue I brought up though.

I've added a test with the global context having a different flag key and variants than the domain scoped one. The domain and global provider context evaluators both listen to `user === 'bob@flags.com'`, but now only the global context flag variant changes instead of both global and domain scoped.

When the providers are the same (same flag keys and variants) they both update when global context is nested under domain context. Did I find a bug or is this desired behaviour?

Reply:
I'm not 100% sure I understand your last comment. But I can confirm what @beeme1mr said here:

>  Basically, domain-scoped context is only managed separately if it's explicitly set for the domain. 

Basically, if you don't set a context for domain X, it will use the default context. If you DO set a context for domain X, it uses that context (this is analogous to how providers are bound to domains as well. Does that clear things up?

My point about the option was somewhat separate: just that users might want to update either of these from their component, and either is valid.

Reply:
i think you cleared it up! i think testing is all set then unless we need more cases.

Reply:
Yes, looks that way to me.